src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -47,124 +35,44 @@
 import java.net.URL;
 import java.util.Map;
 
-public class AppiumCommandExecutor implements CommandExecutor {
+public class AppiumCommandExecutor extends HttpCommandExecutor {
 
-    private final URL remoteServer;
-    private final HttpClient client;
-    private final Map<String, AppiumCommandInfo> additionalCommands;
-    private CommandCodec<HttpRequest> commandCodec;
-    private ResponseCodec<HttpResponse> responseCodec;
-    private DriverService service;
+    private final DriverService service;
 
-    /**
-     * Cretes an instance that sends requests and receives responses.
-     * 
-     * @param additionalCommands is the mapped command repository
-     * @param addressOfRemoteServer is the url to connect to the Appium remote/local server
-     * @param httpClientFactory is the http client factory
-     */
-    public AppiumCommandExecutor(Map<String, AppiumCommandInfo> additionalCommands,
-        URL addressOfRemoteServer, HttpClient.Factory httpClientFactory) {
-        checkNotNull(addressOfRemoteServer);
-        remoteServer = addressOfRemoteServer;
-        this.additionalCommands = additionalCommands;
-        this.client = httpClientFactory.createClient(remoteServer);
+    public AppiumCommandExecutor(Map<String, CommandInfo> additionalCommands,
+                                 URL addressOfRemoteServer, HttpClient.Factory httpClientFactory) {
+        super(additionalCommands, addressOfRemoteServer, httpClientFactory);
+        service = null;
```

Comment:
No. It is checked inside this method
https://github.com/TikhomirovSergey/java-client/blob/12cb9d9b6e117274340e8c73f40b0fa7a11396ba/src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java#L64